File: backend/src/main/java/com/mgs/ergomanager/service/HistoryService.java
package com.mgs.ergomanager.service;

import com.mgs.ergomanager.dto.history.HistoryResponse;
import java.util.List;

/**
 * Read only access to the evaluation history.
 */
public interface HistoryService {

    /**
     * Returns the history entries of a client company, newest first.
     *
     * @param companyId identifier of the company
     * @return list of history entries
     */
    List<HistoryResponse> findByCompany(Long companyId);

    /**
     * Returns the history entries of an employee, newest first.
     *
     * @param employeeEmail email of the employee
     * @return list of history entries
     */
    List<HistoryResponse> findByEmployee(String employeeEmail);
}

File: frontend/src/app/features/history/history.service.ts
import { HttpClient } from '@angular/common/http';
import { Injectable, inject } from '@angular/core';
import { Observable } from 'rxjs';

import { environment } from '../../../environments/environment';
import { HistoryResponse } from '../../shared/models/history.model';

/**
 * HTTP access to the /api/histories endpoints.
 */
@Injectable({ providedIn: 'root' })
export class HistoryService {
  private readonly http = inject(HttpClient);

  private readonly baseUrl = `${environment.apiUrl}/histories`;

  /**
   * Reads the history entries of a client company, newest first.
   *
   * @param companyId identifier of the company
   * @returns list of history entries
   */
  findByCompany(companyId: number): Observable<HistoryResponse[]> {
    return this.http.get<HistoryResponse[]>(this.baseUrl, { params: { companyId } });
  }

  /**
   * Reads the history entries of an employee, newest first.
   *
   * @param employeeEmail email of the employee
   * @returns list of history entries
   */
  findByEmployee(employeeEmail: string): Observable<HistoryResponse[]> {
    return this.http.get<HistoryResponse[]>(
      `${this.baseUrl}/employees/${encodeURIComponent(employeeEmail)}`,
    );
  }
}
